File: src/main/c/jromp/examples/sections.c
#include <stdio.h>
#include <omp.h>

int main(int argc, char *argv[]) {
    #pragma omp parallel sections
    {
		#pragma omp section
		{
			printf("Task 1. Running thread %d\n", omp_get_thread_num());
		}

		#pragma omp section
		{
			printf("Task 2. Running thread %d\n", omp_get_thread_num());
		}

		#pragma omp section
		{
			printf("Task 3. Running thread %d\n", omp_get_thread_num());
		}

        #pragma omp section
		{
			printf("Task 4. Running thread %d\n", omp_get_thread_num());
		}
	}
}

File: src/main/java/jromp/examples/Sections.java
package jromp.examples;

import jromp.parallel.Parallel;
import jromp.parallel.builder.SectionBuilder;
import jromp.parallel.var.Variables;

public class Sections {
	public static void main(String[] args) {
		Parallel.defaultConfig()
		        .sections(SectionBuilder.create()
		                                .task((i, vars) -> System.out.println("Task 1. Running thread " + i))
		                                .variables(Variables.create())
		                                .add()
		                                .task((i, vars) -> System.out.println("Task 2. Running thread " + i))
		                                .variables(Variables.create())
		                                .add()
		                                .task((i, vars) -> System.out.println("Task 3. Running thread " + i))
		                                .variables(Variables.create())
		                                .add()
		                                .task((i, vars) -> System.out.println("Task 4. Running thread " + i))
		                                .variables(Variables.create())
		                                .add()
		        )
		        .join();
	}
}
